src/world/entity/object_handler.cpp

```diff
@@ -9,31 +8,37 @@ namespace world {
 
 namespace entity {
 
+// TODO change to return a pointer (could be null)
 ObjectData&
 ObjectHandler::get_object(const std::string& id) {
     return ided_objects.at(id);
 }
 
 void
-ObjectHandler::read_object(std::filesystem::path object_path) {
+ObjectHandler::read_object(const Manifest::descriptor_t& descriptor) {
     // json to read data into
-    Json::Value object_json;
+
+    object_t object_data;
 
     // read contents from path
-    auto contents = files::open_data_file(object_path);
+    auto contents = files::open_data_file(descriptor.path);
     if (contents.has_value()) {
-        contents.value() >> object_json;
+        std::string content;
+        *contents.value() >> content;
```

Comment:
```suggestion
        *contents >> content;
```

Either use `*` or `.value()`, but not both